Combine VisualizationTools kata with Qubit kata (#495)
microsoft/qsharp: katas/content/qubit/index.md

Several ket symbols have a generally accepted use, such as:

<table>
    <col>

>     <col>

Reviewer: I believe this `<col>` tags are not longer needed after removing the `width` attribute.

@[exercise]({
    "id": "learn_basis_state_amplitudes",
    "title": "Learn Basis State Amplitudes",
    "descriptionPath": "./learn_basis_state_amplitudes/index.md",
    "codePaths": [
        "../KatasLibrary.qs",
        "./learn_basis_state_amplitudes/Verification.qs"
    ],
    "placeholderSourcePath": "./learn_basis_state_amplitudes/Placeholder.qs",
    "solutionPath": "./learn_basis_state_amplitudes/solution.md"
})

@[section]({
    "id": "configure_dumpmachine_output",
    "title": "Configure DumpMachine Output"
})

Sometimes you want to focus on certain properties of the quantum state, such as the relative phases of individual basis states, or on certain parts of the state, such as the basis states with larger amplitudes associated with them. 
[%config](https://docs.microsoft.com/qsharp/api/iqsharp-magic/config) magic command allows you to tweak the format of `DumpMachine` output (available in Q# Jupyter Notebooks only). 
Here are some useful options that help you extract the right information:

* `dump.basisStateLabelingConvention` sets the way the basis states are labeled in the output. Here is how the basis state $|10\rangle$ will look like with different settings:

>   <table style="background-color: white; border:1px solid; tr  { background-color:transparent; }">

Reviewer: Since styling is the responsibility of the component that renders this content, could you remove styling and fixed column widths?

    <col>
    <col width=300>
    <tr>
        <th style="text-align:center; border:1px solid">dump.basisStateLabelingConvention value</th>

>         <th style="text-align:center; border:1px solid">DumpMachine output (first two columns)</th>

Reviewer: Images are not supported in the current way to compose katas, could you create a text version of the content in the second column?

* `dump.measurementDisplayStyle` and `dump.phaseDisplayStyle` set the style of display for "Measurement Probability" and "Phase" columns; you can choose between bar/arrow, number, both, or none for each of the columns. 
If you choose to display the numeric value of measurement probability, you can configure its probability using `dump.measurementDisplayPrecision`.

@[exercise]({
    "id": "high_probability_basis_states",
    "title": "High Probability Basis States",
    "descriptionPath": "./high_probability_basis_states/index.md",
    "codePaths": [
        "../KatasLibrary.qs",
        "./high_probability_basis_states/Verification.qs"
    ],
    "placeholderSourcePath": "./high_probability_basis_states/Placeholder.qs",
    "solutionPath": "./high_probability_basis_states/solution.md"
})

@[section]({
    "id": "display_the_matrix",
    "title": "Display the matrix implemented by the operation"
})

Let's consider a more complex scenario: you've written a Q# operation and want to check that it implements exactly the matrix you're looking for. 

> (An examples of such cases could be quantum oracle implementation, unitary gate synthesis, or more uncommon scenarios such as the [UnitaryPatterns kata](../../UnitaryPatterns/UnitaryPatterns.ipynb)).

Reviewer: ```suggestion
An example of such cases could be quantum oracle implementation and unitary gate synthesis.
```

Reviewer: Removing the reference to a kata that has not been ported yet.

You could do this semi-manually, by applying the operation to each basis state in turn and checking the amplitudes of the resulting states. 
However, [DumpOperation](https://docs.microsoft.com/qsharp/api/qsharp/microsoft.quantum.diagnostics.dumpoperation) library operation offers you a shorter and more elegant way to do this.

> `DumpOperation` is available both in standalone Q# applications as well, but the outputs it produces differs slightly: in plain text mode it is printed with the real and the imaginary components separated into two matrices.

Similarly to `DumpMachine`, the output of `DumpOperation` is accurate up to a global phase.

### Demo: Using DumpOperation

`DumpOperation` requires an operation that acts on an array of qubits. 
If your operation acts on a single qubit, such as most intrinsic gates, or on a mix of individual qubits and qubit arrays, you'll need to write a wrapper for it that takes a single array of qubits as an argument and applies the operation to the right qubits of that array.

@[example]({"id": "dump_operation_demo", "codePath": "./examples/DumpOperationDemo.qs"})

> > Note that the matrix of the CNOT gate produced by `DumpOperation` in this demo might differ from the one you are used to seeing in quantum computing resources, such as the [multi-qubit gates tutorial](../MultiQubitGates/MultiQubitGates.ipynb#CNOT-Gate):

Reviewer: Since we don't have notebook files to include as references, for now we can use the ID of the kata or exercise that is being referenced.
